core/src/main/java/com/linecorp/armeria/server/annotation/ResponseConverterFunction.java

```diff
@@ -75,4 +75,23 @@ static <T> T fallthrough() {
         // Always throw the exception quietly.
         throw FallthroughException.get();
     }
+
+    /**
+     * Only return the defaultResponseConverterFunction if it is not the same class or a superclass of
+     * the current instance
+     * @param defaultResponseConverterFunction
+     * @return
+     */
+    default ResponseConverterFunction orElse(ResponseConverterFunction defaultResponseConverterFunction) {
+        // The code below has been commented out because the ResponseConverterFunction passed in is the
+        // CompositeResponseConverterFunction, which contains a list of backingConverters
+
+        /* if (defaultResponseConverterFunction.getClass().isInstance(this)) {
+            return this;
+        } else {
+            return defaultResponseConverterFunction;
+        }*/
+
+        return this;
+    }
```

Comment:
We don't really know at this point (given for example, a `CompositeResponseConverterFunction`) whether the passed-in function is capable of converting.

Reply:
That's a good question. If the `ResponseConverterFunction` is capable of converting, it returns an `HttpResponse`. If it's not it throw a `FallthroughException` as illustrated here: https://github.com/line/armeria/pull/4312/files#diff-bf99df43890c22eb9efdf47e3ea210cdb632117b76c439c4ceb6f7af49904745R70-R76

So we need to try this and the other:
```java
default ResponseConverterFunction orElse(ResponseConverterFunction other) {
    return new OrElseResponseConverterFunction(this, other);
}

final class OrElseResponseConverterFunction implements ResponseConverterFunction {

    private final ResponseConverterFunction first;
    private final ResponseConverterFunction second;

    OrElseResponseConverterFunction(ResponseConverterFunction first,
                                    ResponseConverterFunction second) {
        this.first = first;
        this.second = second;
    }

    @Override
    public HttpResponse convertResponse(ServiceRequestContext ctx, ResponseHeaders headers,
                                        @Nullable Object result, HttpHeaders trailers) throws Exception {
        try {
            return first.convertResponse(ctx, headers, result, trailers);
        } catch (FallthroughException ignored) {
            return second.convertResponse(ctx, headers, result, trailers);
        }
    }
}


```